pkg/xds/context/mesh_context_builder.go

```diff
@@ -284,19 +287,6 @@ func (m *meshContextBuilder) fetchResourceList(ctx context.Context, resType core
 		return nil, fmt.Errorf("unknown resource scope:%s", desc.Scope)
 	}
 
-	// For some resources we apply extra filters
-	switch resType {
-	case core_mesh.ServiceInsightType:
-		if mesh == nil {
-			return desc.NewList(), nil
-		}
-		// ServiceInsights in XDS generation are only used to check whether the destination is ready to receive mTLS traffic.
-		// This information is only useful when mTLS is enabled with PERMISSIVE mode.
-		// Not including this into mesh hash for other cases saves us unnecessary XDS config generations.
-		if backend := mesh.GetEnabledCertificateAuthorityBackend(); backend == nil || backend.Mode == mesh_proto.CertificateAuthorityBackend_STRICT {
-			return desc.NewList(), nil
-		}
-	}
```

Comment:
There's a performance impact here we should discuss. My biggest concern is that service-insight gets updated whenever a DPP goes up or down. That's potentially a lot of very frequent updates

Reply:
Yes, that's true. I think it is a potential blocker for using ServiceInsights for protocol map. Other options is to use `EndpointMap` in the mesh context and calculate by ourselves